.github/workflows/build-deploy.yml

```
@@ -26,74 +30,147 @@ jobs:
 
-      - name: Configure AWS credentials
-        uses: aws-actions/configure-aws-credentials@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2.1.0
         with:
-          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-          aws-region: ${{ secrets.AWS_REGION }}
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Login to Amazon ECR
-        id: login-ecr
-        uses: aws-actions/amazon-ecr-login@v1
+      # This step outputs Docker labels and Docker tag associated with github.ref (branch name in this scenario)
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v4.3.0
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
 
-      - name: Build & push docker images
-        env:
-          BUILD_ID: ${{needs.unique_id.outputs.unique_id}}
-          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+      # When running on develop branch we want to tag Docker images with unique id and branch name only.
+      # When running on master and release we're adding sem ver and latest (for release) tags too.
+      - name: Setup Docker tags
+        id: tags
         run: |
-          docker build --build-arg SENTRY_RELEASE="${{needs.unique_id.outputs.unique_id}}" -t $ECR_REGISTRY/ics-api:$BUILD_ID -f docker/prod/Dockerfile .
-          docker push $ECR_REGISTRY/ics-api:$BUILD_ID
-          docker build --build-arg PUBLIC_IMAGE=$ECR_REGISTRY/ics-api:$BUILD_ID --build-arg SENTRY_COMMIT="$GITHUB_SHA" --build-arg SENTRY_RELEASE="${{needs.unique_id.outputs.unique_id}}" --build-arg SENTRY_AUTH_TOKEN=${{ secrets.SENTRY_AUTH_TOKEN }} -f docker/sentry-upload/Dockerfile .
-      - name: Logout of Amazon ECR
-        if: always()
-        run: docker logout ${{ steps.login-ecr.outputs.registry }}
+              if [ "${{ github.ref }}" = "refs/heads/develop" ]; then
+                echo "TAGS=${{ steps.meta.outputs.tags }},${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{  needs.configure.outputs.IMAGE_TAG  }}" >> $GITHUB_OUTPUT
+              else
+                echo "TAGS=${{ steps.meta.outputs.tags }},${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{  needs.configure.outputs.IMAGE_TAG  }},${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{  needs.tag-release.outputs.tag  }}" >> $GITHUB_OUTPUT
+              fi
+              
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          push: ${{  needs.configure.outputs.PUSH_DOCKER  }}
+          platforms: linux/amd64,linux/arm64
+          tags: ${{ steps.tags.outputs.TAGS }}
+          labels: ${{ steps.meta.outputs.labels }}
+          file: ./docker/Dockerfile
```

This section needs to be tweaked as we've got two docker builds to perform, one from `docker/prod/Dockerfile` and one from `docker/sentry-upload/Dockerfile`

---

Maybe all we need to do is add another step that uploads the sentry images. 

We also need to pass the build arguments

```suggestion
      - name: Build and push Docker image
        uses: docker/build-push-action@v3
        with:
          context: .
          push: ${{  needs.configure.outputs.PUSH_DOCKER  }}
          platforms: linux/amd64,linux/arm64
          tags: ${{ steps.tags.outputs.TAGS }}
          labels: ${{ steps.meta.outputs.labels }}
          file: ./docker/prod/Dockerfile
          build-args: |
            SENTRY_RELEASE="${{ needs.configure.outputs.IMAGE_TAG }}"
          
     # Sentry upload is done via a docker build in order to have access to the app files in the image
     - name: create Sentry release
        uses: docker/build-push-action@v3
        with:
          context: .
          push: false
          file ./docker/sentry-upload/Dockerfile
          build-args: |
```